### ISSUE TITLE
Inconsistent query behaviour for fixed length variable length path queries

### ISSUE THREAD

**Issue by gkorland** [FalkorDB/FalkorDB#73]
Created by: @K-Jo
Source: https://github.com/RedisGraph/RedisGraph/issues/321
TL;DR: RedisGraph returns DISTINCT results for _fixed length_ `variable length path` queries with a single starting point, **without** mentioning the DISTINCT keyword.

Given
```
CREATE (n0:A {name: 'n0'}),
       (n00:B {name: 'n00'}),
       (n01:B {name: 'n01'}),
       (n000:C {name: 'n000'}),
       (n0)-[:LIKES]->(n00),
       (n0)-[:LIKES]->(n01),
       (n00)-[:LIKES]->(n000),
       (n01)-[:LIKES]->(n000)
```
The following query
```
MATCH (a:A)-[:LIKES*2]->(c) WHERE a.name='n0' RETURN c
```
Should return **twice** the node `n000`.  However RedisGraph returns `n000` only **once**
```
redis:6379> graph.query test "MATCH (a:A)-[:LIKES*2]->(c) WHERE a.name='n0' RETURN c"
1) 1) 1) "c.name"
   2) 1) "n000"
```

This bug can be explained via the implementation of RedisGraph.  The query will execute a _conditional traverse_ for each node with the label A.  For each node, RedisGraph will create a Hyper Sparse Vector with a single element and multiply it twice by the adjacency matrix of relationships of type LIKES.  The resulting vector will have a Non Zero Value (NZV) at node `n000`, however it did not keep track of the actual paths that lead to it. 

In such, the result of this query is incorrect and behaves as
```
MATCH (a:A)-[:LIKES*2]->(c) WHERE a.name='n0' RETURN DISTINCT c
```

**Note 1:  The behaviour is correct for (_non fixed_) `variable length path` queries. e.g.**
```
MATCH (a:A)-[:LIKES*2..3]->(c) WHERE a.name='n0' RETURN c
```
returns correctly `n000` twice
```
redis:6379> graph.query test "MATCH (a:A)-[:LIKES*2..3]->(c) WHERE a.name='n0' RETURN c"
1) 1) 1) "c.name"
   2) 1) "n000"
   3) 1) "n000"
```
**Note 2: For more than 1 starting point, a  _fixed length_ `variable length path` query returns non distinct values if it the destination node could be reached from different starting points.** 
```
MATCH (b:B)-[:LIKES*1]->(c)  RETURN c
```
This query should return `n000` twice since there are two start nodes with the label B.
```
redis:6379> graph.query test "MATCH (b:B)-[:LIKES*1]->(c)  RETURN c"
1) 1) 1) "c.name"
   2) 1) "n000"
   3) 1) "n000"
```

**Comment by DvirDukhan** [comment]
Closing as no longer reproducible.

This was re-tested on the current stable release, **v4.20.1**
(`falkordb/falkordb:latest`), and the reported behaviour no longer occurs:

**Reproduction**

```cypher
CREATE (n0:A {name: 'n0'}), (n00:B {name: 'n00'}), (n01:B {name: 'n01'}), (n000:C {name: 'n000'}), (n0)-[:LIKES]->(n00), (n0)-[:LIKES]->(n01), (n00)-[:LIKES]->(n000), (n01)-[:LIKES]->(n000)

MATCH (a:A)-[:LIKES*2]->(c) WHERE a.name='n0' RETURN c.name

MATCH (a:A)-[:LIKES*2..3]->(c) WHERE a.name='n0' RETURN c.name

MATCH (b:B)-[:LIKES*1]->(c) RETURN c.name
```

| Build | Result |
| --- | --- |
| `falkordb/falkordb:latest` (v4.20.1) | MATCH (a:A)-[:LIKES*2]->(c) WHERE a.name='n0' RETURN c.name correctly returns 'n000' twice (once per path); MATCH (b:B)-[:LIKES*1]->(c) RETURN c.name also correctly returns 'n000' twice |

Issue is from the original RedisGraph project (2023-era, ported from RedisGraph#321) claiming fixed-length variable-length-path traversals silently deduplicated results as if DISTINCT were applied. Reproduced the exact CREATE + queries from the issue body on v4.20.1 (latest) and the bug is already gone there -- both fixed-length and multi-start-point cases return the expected duplicate rows. edge-rs matches latest exactly. Since latest already behaves correctly, this cannot be credited to the Rust migration.

The defect appears to have been fixed in the C engine at some point between the
version this was reported against and v4.20.1. If you can still reproduce it on
v4.20.1 or later, please reopen with the exact version and a repro.

**Comment by gkorland** [comment]
Regression test added in #2471 (`tests/flow/test_variable_length_traversals.py::test18_fixed_length_path_multiplicity`), pinning path multiplicity for fixed-length variable-length traversals. Verified the assertion fails on `v4.2.2` (returns the destination once) and passes on the current Rust engine (twice), including the with/without path-variable equivalence.
